metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -716,6 +722,47 @@ class MetalsLanguageServer(
     onChange(paths).asJava
   }
 
+  private def insertPackage(path: AbsolutePath): Unit = {
+    def createPackageName(path: AbsolutePath): Option[String] = {
+      if (path.toFile.length() == 0) {
+        val sourceDirectory = buildTargets.inverseSourceDirectory(path)
+        val relativeDirectory = sourceDirectory
+          .map(path.toRelative)
+          .flatMap(relativePath => Option(relativePath.toFile.getParentFile))
+          .map(_.getPath)
+        val packageName = relativeDirectory.map(_.replace("/", "."))
+
+        packageName.map("package " + _)
+      } else {
+        None
+      }
+    }
+
+    def createPackageWorkspaceEdit(packageName: String): WorkspaceEdit = {
```

Comment:
could we drop the word `package` ?  We are already in the context of creating a package-related edit so I think there is no need to repeat it here
```suggestion
    def createWorkspaceEdit(packageName: String): WorkspaceEdit = {
```